src/main/java/org/mskcc/oncokb/curation/util/CdxUtils.java

```diff
@@ -0,0 +1,227 @@
+                }
+            }
+
+            if (targetTable == null) {
+                log.warn("No information found for fda submission : {}", code);
+                continue;
+            }
+
+            Element tbody = targetTable.select("tbody").first();
+
+            FdaSubmission fdaSubmission = new FdaSubmission();
+
+            for (Element tr : tbody.getElementsByTag("tr")) {
+                Element th = tr.getElementsByTag("th").first();
+                Element td = tr.getElementsByTag("td").first();
+
+                if (th != null) {
+                    String header = th.text().trim();
+                    String content = td.text().trim();
+                    if (this.cdxDeviceFieldNames.stream().anyMatch(header::equalsIgnoreCase)) {
+                        fdaSubmission.setDeviceName(content);
+                    } else if (this.cdxNumberFieldNames.stream().anyMatch(header::equalsIgnoreCase)) {
+                        fdaSubmission.setNumber(content);
+                    } else if (header.equalsIgnoreCase("Generic Name")) {
+                        fdaSubmission.setGenericName(content);
+                    } else if (header.equalsIgnoreCase("Supplement Number")) {
+                        fdaSubmission.setSupplementNumber(content);
+                    } else if (header.equalsIgnoreCase("Date Received")) {
+                        fdaSubmission.setDateReceived(convertDateToInstant(content));
+                    } else if (header.equalsIgnoreCase("Decision Date")) {
+                        fdaSubmission.setDecisionDate(convertDateToInstant(content));
+                    }
+                } else {
+                    // The Approval Order Statement column does not have the header stored in <th> element
+                    // Instead it is a <span> inside <td> element.
+                    th = td.children().first();
+                    if (th != null) {
+                        String header = th.text().trim();
+                        if (header.equalsIgnoreCase("Approval Order Statement")) {
+                            String approvalOrderStatement = td.textNodes().get(td.textNodes().size() - 1).text();
+                            fdaSubmission.setDescription(approvalOrderStatement);
+                        }
+                    }
+                }
+            }
+
+            // Add FDASubmissionType
+            if (fdaSubmission.getNumber() != null) {
+                Optional<FdaSubmissionType> type = fdaSubmissionTypeService.findOneBySubmissionNumber(fdaSubmission.getNumber());
+                if (type.isPresent()) {
+                    fdaSubmission.setType(type.get());
+                }
+            }
+            fdaSubmissions.add(fdaSubmission);
+        }
+        return fdaSubmissions;
+    }
+
+    private Instant convertDateToInstant(String date) {
+        return LocalDate.parse(date, DateTimeFormatter.ofPattern("MM/dd/uuuu", Locale.US)).atStartOfDay().toInstant(ZoneOffset.UTC);
```

Comment:
This is interesting, I always use `yyyy`. Good to know! https://stackoverflow.com/questions/41177442/uuuu-versus-yyyy-in-datetimeformatter-formatting-pattern-codes-in-java